src/utils/Shell.ts

```diff
@@ -78,4 +113,8 @@ const {shell} = new class {
     };
 };
 
+export function loadAllHistories(): { lastModified: Date, commands: string[] }[] {
```

Comment:
You can safely assume that a person only wants to read history from his login shell, which is `loginShell`. Also, it'll be consistent with alias loading behaviour. So, please remove this function and use `loginShell. loadHistory()` instead.